api/src/test/java/marquez/api/OpenLineageResourceTest.java

```diff
@@ -0,0 +1,67 @@
+package marquez.api;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.collect.ImmutableSortedSet;
+import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
+import io.dropwizard.testing.junit5.ResourceExtension;
+import java.util.Map;
+import marquez.common.Utils;
+import marquez.service.LineageService;
+import marquez.service.ServiceFactory;
+import marquez.service.models.Lineage;
+import marquez.service.models.Node;
+import marquez.service.models.NodeId;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+class OpenLineageResourceTest {
+  private static ResourceExtension UNDER_TEST;
+
+  static {
+    LineageService lineageService = mock(LineageService.class);
+
+    Node testNode =
+        Utils.fromJson(
+            OpenLineageResourceTest.class.getResourceAsStream("/lineage/node.json"),
+            new TypeReference<>() {});
+
+    when(lineageService.lineage(any(NodeId.class), anyInt()))
+        .thenReturn(new Lineage(ImmutableSortedSet.of(testNode)));
+
+    ServiceFactory serviceFactory =
+        ApiTestUtils.mockServiceFactory(Map.of(LineageService.class, lineageService));
+
+    UNDER_TEST =
+        ResourceExtension.builder()
+            .addResource(new OpenLineageResource(serviceFactory))
+            .addResource(new LineageResource(serviceFactory))
+            .build();
+  }
+
+  @Test
+  void testEqualResponseFromV1AndV1Beta() {
+    Lineage responseBeta =
+        UNDER_TEST
+            .target("/api/v1-beta/lineage")
+            .queryParam("nodeId", "job:test")
+            .request()
+            .get()
+            .readEntity(Lineage.class);
+    Lineage response =
+        UNDER_TEST
+            .target("/api/v1/lineage")
+            .queryParam("nodeId", "job:test")
```

Comment:
Do we know what the expect response would be? A node ID can either be `dataset:<namespace>:<dataset>` or `job:<namespace>:<job>`. So, I _think_ the node ID `job:test` used in your test should throw a `500`.

Reply:
The service mock on line 35 will return the contents of `node.json` no matter what its inputs are

Reply:
👍

Reply:
so, value  "job:test" passes pattern matching in NodeId
  ```
private static final Pattern ID_PATTERN =
      Pattern.compile(
          String.format("^(%s|%s|%s):.*$", ID_PREFX_DATASET, ID_PREFX_JOB, ID_PREFX_RUN));
```
I guess we should rethink that pattern, cause we will fail later on.

```
@JsonIgnore
  public JobVersionId asJobVersionId() {
    String[] parts = parts(4, ID_PREFX_JOB);
    return new JobVersionId(
        NamespaceName.of(parts[1]), JobName.of(parts[2]), UUID.fromString(parts[3]));
  }

  @JsonIgnore
  public DatasetVersionId asDatasetVersionId() {
    String[] parts = parts(4, ID_PREFX_DATASET);
    return new DatasetVersionId(
        NamespaceName.of(parts[1]), DatasetName.of(parts[2]), UUID.fromString(parts[3]));
  }
```